### PR TITLE
Add sqrt and is_square to snarkette

### PR DESCRIPTION
**Pull request by imeckler**
This adds `sqrt` and `is_square` to snarkette. I copied the Tonelli--Shanks implementation from `libff/algebra/fields/fp.tcc`.  I wrote a few unit tests to check these functions.

### CODE REVIEW

File: src/lib/snarkette/fields.ml [hunk 1]
```diff
@@ -135,7 +141,24 @@ module Make_fp
 
   let ( * ) x y = x * y % Info.order
 
-  let square x = x * x
+  let square x = x * x % Info.order
```
Comment (cmr) [inline src/lib/snarkette/fields.ml]: Isn't this `%` redundant? The definition of `*` right above includes it.

File: src/lib/snarkette/fields.ml [hunk 2]
```diff
@@ -144,6 +167,87 @@ module Make_fp
...
+          ~init:{z; b; x; v}
+          (fun {z; b; x; v} ->
+            let m = pow2_order b in
+            let w = pow2 z Int.(v - m - 1) in
+            let z = square w in
+            {z; b= b * z; x= x * w; v= m} )
+      in
+      x
+
+  let%test_unit "sqrt test" =
```
Comment (cmr) [inline src/lib/snarkette/fields.ml]: Can you make this a quickcheck?

File: src/lib/snarkette/fields.ml (hunk 1 above)
Comment (imeckler) [reply src/lib/snarkette/fields.ml]: Yes thank you!

**Comment by imeckler** [comment]
Okie doke @cmr, both done

**Comment by cmr** [comment]
It seems you need to turn down the number of iterations in the quickcheck.

**Comment by imeckler** [comment]
> It seems you need to turn down the number of iterations in the quickcheck.

Thank you

**Pull request merged**